### PR TITLE
MRG: Fix plot_compare_evokeds to support 'misc' channel type

### LINKED ISSUE #3963: How to define channel types on the fly
In my lab we use multiple synced signals (EEG, eye-tracking, ...), and one of them is information of spatial markers (ndigital optotrak). I create RawArrays from the spatial data with 'misc' channel types. The problem is that every time I want to plot the evoked data a __ValueError: Non-data channel picked.__ is raised. My temporary (lazy) solution is to change the channels types to a valid one (as those defined in _DATA_CH_TYPES_SPLIT), but it is far from a good solution. So it would be nice if I could define a channel type (with its units to plot, etc) when using other ___non-standard___ data types (e.g. eye-tracking/mouse-tracking channels in pixels or mm, spatial markers in mm, among others) while profiting of this great software.

### PR DESCRIPTION
**Pull request by cmmoenne**
Fixes #3963 

to allow use 'misc' channels in plot_compare_evokeds. I replace _DATA_CH_TYPES_SPLIT (defined as ['mag', 'grad', 'eeg', 'seeg', 'ecog', 'hbo', 'hbr'] [see here](https://github.com/mne-tools/mne-python/blob/master/mne/io/pick.py#L715)) with the same valid types defined in [_plot_evoked](https://github.com/mne-tools/mne-python/blob/master/mne/viz/evoked.py#L200).

### CODE REVIEW

**Comment by jona-sassenhagen** [comment]
Thanks for taking care of this ... Sorry, I'm sick and I have 100s of emails of backlog :(

**Comment by cmmoenne** [comment]
no problem! I hope you feel better soon :-)

File: mne/viz/evoked.py [hunk 1]
```diff
@@ -1472,7 +1472,11 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
     ch_types = list(set(channel_type(example.info, pick_)
                     for pick_ in picks))
     # XXX: could possibly be refactored; plot_joint is doing a similar thing
-    if any([type_ not in _DATA_CH_TYPES_SPLIT for type_ in ch_types]):
+    # Valid data types ordered for consistency (same as _plot_evoked)
+    valid_channel_types = ['eeg', 'grad', 'mag', 'seeg', 'eog', 'ecg', 'emg',
+                           'dipole', 'gof', 'bio', 'ecog', 'hbo', 'hbr',
+                           'misc']
```
Comment (agramfort) [inline mne/viz/evoked.py]: I would reuse _DATA_CH_TYPES_SPLIT and just add misc to it

valid_channel_types =_DATA_CH_TYPES_SPLIT + ['misc']

Comment (jaeilepp) [inline mne/viz/evoked.py]: That would exclude ``'eog', 'ecg', 'emg', 'dipole', 'gof', 'bio'``. Maybe there should be ``_VALID_CHANNEL_TYPES`` for ``viz/evoked``, so both plotters use the same channel types (and reduce maintenance).

Comment (cmmoenne) [inline mne/viz/evoked.py]: So which way we do take? 

From a design point of view I would use 'picks' or 'exclude' parameters to filter and let the user plot whatever they want. For instance, the default of 'exclude' could be 'valid' which means valid channel types (defined in io.pick) minus 'bads' channels to behave as it is now.

Comment (Eric89GXL) [inline mne/viz/evoked.py]: Sorry for the delay @cmmoenne. Yeah I would say plot whatever is available if possible. I think that's what @jaeilepp might have been getting at in terms of the `_VALID_CHANNEL_TYPES`, a list of channel types that we support across plotting functions. Although maybe that would just be all of them, in which case we already have such a list.

Is that what you were thinking @jaeilepp?

Comment (jaeilepp) [inline mne/viz/evoked.py]: Yeah, using all channels probably shouldn't cause any problems since a lot of them are dropped at epoching stage anyway.

Comment (cmmoenne) [inline mne/viz/evoked.py]: If that's the case, the best would be to check against the already defined list of supported channels. @Eric89GXL Where is that list? I didn't see anything like that in io.pick.

Comment (Eric89GXL) [inline mne/viz/evoked.py]: I thought we had one, but if it's not there we probably don't. We might need to create it.

Comment (cmmoenne) [inline mne/viz/evoked.py]: @Eric89GXL I made some changes based on the feedback (I used the @jaeilepp idea of a `_VALID_CHANNEL_TYPES` in viz/evoked).

**Comment by agramfort** [comment]
LGTM

**Comment by Eric89GXL** [comment]
@cmmoenne does this fix your plotting case?

File: mne/viz/evoked.py [hunk 2]
```diff
@@ -33,6 +33,12 @@
                                _auto_topomap_coords)
 
 
+# Valid data types ordered for consistency
+_VALID_CHANNEL_TYPES = ['eeg', 'grad', 'mag', 'seeg', 'eog', 'ecg', 'emg',
+                        'dipole', 'gof', 'bio', 'ecog', 'hbo', 'hbr',
+                        'misc']
```
Comment (Eric89GXL) [inline mne/viz/evoked.py]: This should probably live in `..io.pick` instead (we'll want to add entries to it if we add new channel types)

**Comment by cmmoenne** [comment]
@Eric89GXL  Yes, it solves my problem.

However, it is worth noting that there is a border case of use. When using `picks`'s default value (*list()*), gfp is set to *True* and `_pick_data_channels` is used to determine data channels. As `_pick_data_channels` don't pick *misc* channels by default, `plot_compare_evokeds` will raise an exception when checking the number of channels to perform a GFP. Specifically, `_pick_data_channels` use a `pick_types` call with `_PICK_TYPES_DATA_DICT` as hardcoded parameter. Misc channels will never be able to be plotted, because `_PICK_TYPES_DATA_DICT` defines misc = *False*.

I don't know if GFP has some meaning when working with misc data, and I think that change `_PICK_TYPES_DATA_DICT` is too risky for such a small profit (or no profit at all). We have 2 possibilities if we want to be nice with the user:

**a)** raise an exception earlier: 'GFP not supported with ...'.

**b)** or, as the user can bypass the error by selecting channels with picks, call `pick_types` directly with a copy of `_PICK_TYPES_DATA_DICT` that sets misc to *True* (and other valid channels of `_VALID_CHANNEL_TYPES` that are also *False* in `_PICK_TYPES_DATA_DICT`). Alternatively, we can do something similar by changing the signature of `_pick_data_channels` with a default parameter `pick_types=_PICK_TYPES_DATA_DICT`, allowing change the picked types if required.

(finally, the out of the scope option)
**c)** As I use `picks` extensively in my 'misc' datasets, I suspect that a bunch of functions raise exceptions when using misc channels (don't remember which ones, but it happened to me). However, by using picks directly the errors don't raise. Perhaps what needs to be fixed lies beyond this PR.

...also, I move `_VALID_CHANNEL_TYPES` to io.pick in my last commit.

**Comment by Eric89GXL** [comment]
My vote would be to keep it simple and just do a) raise an error for now, saying to set `gfp=False`. We can always add the functionality in b) later if users feel like the GFP is useful/important for them in such cases.

You can add the error to this PR if you want

**Comment by cmmoenne** [comment]
PR updated.

Now an error is thrown when `picks` is empty and no valid channels are found. Note that picking channels manually let the user plot the gfp of non-valid channels, such as misc, when `gfp=True`.

**Comment by Eric89GXL** [comment]
Is this good to go from your end? If so please prefix `MRG` to the title

**Comment by agramfort** [comment]
LGTM

**Comment by jaeilepp** [comment]
Thanks @cmmoenne

**Pull request merged**